src/Type/ObjectStorageDynamicReturnTypeExtension.php

```diff
@@ -41,7 +41,7 @@ public function getTypeFromMethodCall(
             $propertyName = lcfirst(substr($methodCall->var->name, 3));
 
             $class = $scope->getClassReflection();
-            if ($class->hasProperty($propertyName)) {
+            if ($class->hasProperty($propertyName)->yes()) {
```

Comment:
This is also wrong. `ClassReflection::hasProperty()` still returns bool. Did you not bother with testing your code?